### ISSUE TITLE
Вопрос по поводу автозапуска

### ISSUE THREAD

**Issue by leonelo32** [bol-van/zapret#459]
Не очень удобно при каждом запуске windows видеть окно запуска от имени администратора, а потом всё время смотреть на окно консоли zapret, поэтому меня волнует вопрос, можно ли сделать автозапуск zapret в виде службы? Просто сейчас это не очень удобно

**Comment by iRadioMan** [comment]
Да, редактируете файл service_create.cmd (если необходимо прописать настройки), запускаете от администратора, он создаст службу. Затем запускаете службу через service_start.cmd

**Comment by DuhlesGO** [comment]
Можете помочь перенести настройки из preset_discord в service_create для запуска служба с теми параметрами из preset_discord
[service.zip](https://github.com/user-attachments/files/17307514/service.zip)

**Comment by Aeasur** [comment]
> Да, редактируете файл service_create.cmd (если необходимо прописать настройки), запускаете от администратора, он создаст службу. Затем запускаете службу через service_start.cmd

Если не сложно, объясните подробнее что нужно редактировать в service_start.cmd. К примеру у меня zapret прекрасно работает при запуске preset_russia.cmd в ручную, но скопировав из него аргументы в service_start.cmd ничего не происходит.

**Comment by Vadim910** [comment]
> > Да, редактируете файл service_create.cmd (если необходимо прописать настройки), запускаете от администратора, он создаст службу. Затем запускаете службу через service_start.cmd
> 
> Если не сложно, объясните подробнее что нужно редактировать в service_create.cmd. К примеру у меня zapret прекрасно работает при запуске preset_russia.cmd в ручную, но скопировав из него аргументы в service_create.cmd ничего не происходит.

ну смотри, тебе надо заменить аргументы, установленные по умолчанию, на аргументы, которые тебе выдал Blockcheck.cmd. допустим в моем случае это: ` --wf-l3=ipv4 --wf-tcp=80 --dpi-desync=disorder2, --wf-l3=ipv4 --wf-tcp=443 --dpi-desync=disorder2, --wf-l3=ipv4 --wf-udp=443 --dpi-desync=fake, --wf-l3=ipv4 --wf-tcp=443 --dpi-desync=fake --dpi-desync-ttl=4 `
теперь я открываю через блокнот файл service_create.cmd. изначально там:
```
rem set ARGS=--wf-l3=ipv4,ipv6 --wf-tcp=80,443 --dpi-desync=fake,split --dpi-desync-ttl=7 --dpi-desync-fooling=md5sig
rem call :srvinst winws1
rem set ARGS=--wf-l3=ipv4,ipv6 --wf-udp=443 --dpi-desync=fake 
rem call :srvinst winws2
goto :eof 

:srvinst
net stop %1
sc delete %1
sc create %1 binPath= "\"%~dp0winws.exe\" %ARGS%" DisplayName= "zapret DPI bypass : %1" start= auto
sc description %1 "zapret DPI bypass software"
sc start %1 
```
а тебе нужно заменить аргументы, повторюсь, на аргументы, которые Blockcheck.cmd выдал. у меня получилось:
```
@echo off
:: Устанавливаем аргументы для первой службы winws1
set ARGS=--wf-l3=ipv4 --wf-tcp=80 --dpi-desync=disorder2
call :srvinst winws1

:: Устанавливаем аргументы для второй службы winws2
set ARGS=--wf-l3=ipv4 --wf-tcp=443 --dpi-desync=disorder2
call :srvinst winws2

:: Устанавливаем аргументы для третьей службы winws3
set ARGS=--wf-l3=ipv4 --wf-udp=443 --dpi-desync=fake
call :srvinst winws3

:: Устанавливаем аргументы для четвертой службы winws4
set ARGS=--wf-l3=ipv4 --wf-tcp=443 --dpi-desync=fake --dpi-desync-ttl=4
call :srvinst winws4

goto :eof

:srvinst
net stop %1
sc delete %1
sc create %1 binPath= "\"%~dp0winws.exe\" %ARGS%" DisplayName= "zapret DPI bypass : %1" start= auto
sc description %1 "zapret DPI bypass software"
sc start %1
```
теперь при каждом запуске компа у меня запускаются все службы и все работает
